Sources/Commands/Options.swift

```diff
@@ -170,9 +170,23 @@ public struct SwiftToolOptions: ParsableArguments {
     var shouldDisableSandbox: Bool = false
 
     /// Disables manifest caching.
-    @Flag(name: .customLong("disable-package-manifest-caching"), help: "Disable caching Package.swift manifests")
+    @Flag(name: .customLong("disable-package-manifest-caching"), help: .hidden)
     var shouldDisableManifestCaching: Bool = false
 
+    /// Disables manifest caching.
+    @Option(name: .customLong("manifest-caching"), help: "Caching of Package.swift manifests")
+    var manifestCachingMode: ManifestCachingMode = .system
+
+    enum ManifestCachingMode: String, ExpressibleByArgument {
+        case none
+        case local
+        case system
+
+        init?(argument: String) {
+            self.init(rawValue: argument)
+        }
+    }
+
```

Comment:
@abertelrud per your idea ^^

Reply:
This makes sense to me.  @neonichu what do you think?

I don't know if swift-argument-parser provides a way of explaining each enum in an option, but we can always add that later if they do.

Reply:
Something was bothering me and the word "system" here, and it just occurred to me what this is:  the shared cache is actually still per user, not system, so I wonder if "shared" is more correct here?  It fits less well with "local" (should it be "package"?), but unless it's actually a truly shared system cache I don't think we should call it system.  Bikeshedding, I know, but name connotations do matter.

Reply:
naming is hard, totally open to 'shared' or any other idea. @neonichu?

Reply:
also, is using 'system'/'shared' by default sounds good?

Reply:
I think "shared" sounds good.